test/integration/src/main/java/io/pravega/test/integration/utils/TestUtils.java

```diff
@@ -0,0 +1,115 @@
+import static org.junit.Assert.assertTrue;
+
+/**
+ * A helper class with general-purpose utility methods for integration tests.
+ */
+@Slf4j
+public class TestUtils {
+
+    public static void createStreams(ClientConfig clientConfig, String scopeName, List<String> streamNames) {
+        @Cleanup
+        StreamManager streamManager = StreamManager.create(clientConfig);
+        assertNotNull(streamManager);
+
+        boolean isScopeCreated = streamManager.createScope(scopeName);
+        assertTrue("Failed to create scope", isScopeCreated);
+
+        streamNames.forEach(s -> {
+            boolean isStreamCreated =
+                    streamManager.createStream(scopeName, s, StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(1)).build());
+            if (!isStreamCreated) {
+                throw new RuntimeException("Failed to create stream: " + s);
+            }
+        });
+    }
+
+    @SneakyThrows
+    public static void writeDataToStream(String scope1, String stream1, String message1, ClientConfig writerClientConfig) {
+        @Cleanup final EventStreamClientFactory writerClientFactory = EventStreamClientFactory.withScope(scope1,
+                writerClientConfig);
+
+        @Cleanup final EventStreamWriter<String> writer = writerClientFactory.createEventWriter(stream1,
+                new JavaSerializer<String>(),
+                EventWriterConfig.builder().build());
+        writer.writeEvent(message1).get();
+        log.info("Wrote a message to the stream {}/{}", scope1, stream1);
+    }
+
+    public static String readAMessageFromStream(String scopeName, String streamName, ClientConfig readerClientConfig,
+                                                 String readerGroupName) {
+        @Cleanup
+        EventStreamClientFactory readerClientFactory = EventStreamClientFactory.withScope(scopeName, readerClientConfig);
+        log.debug("Created the readerClientFactory");
+
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .stream(Stream.of(scopeName, streamName))
+                .disableAutomaticCheckpoints()
+                .build();
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, readerClientConfig);
+        readerGroupManager.createReaderGroup(readerGroupName, readerGroupConfig);
+        log.debug("Created reader group with name {}", readerGroupName);
+
+        @Cleanup
+        EventStreamReader<String> reader = readerClientFactory.createReader(
+                "readerId", readerGroupName,
+                new JavaSerializer<String>(), ReaderConfig.builder().initialAllocationDelay(0).build());
+        log.debug("Created an event reader");
+
+        // Keeping the read timeout large so that there is ample time for reading the event even in
+        // case of abnormal delays in test environments.
+        String readMessage = reader.readNextEvent(10000).getEvent();
```

Comment:
It may surprise you but on Travis, this timeout may not be sufficient. Expect this to be a root cause for random failures.

If the timeout expires, you won't get a `TimeoutException`; rather the `getEvent()` will return null, which will likely manifest as a NullPtrEx upstream.

Reply:
I've increased the timeout to account for that.